cfme/tests/infrastructure/test_providers.py

```diff
@@ -523,3 +530,63 @@ def test_compare_templates(appliance, setup_provider_min_templates, provider, mi
 
     t_list = [t.name for t in t_coll]
     assert compare_view.verify_checked_items_compared(t_list, compare_view)
+
+
+@pytest.fixture
+@pytest.mark.provider([AnsibleTowerProvider], selector=ONE_PER_VERSION, scope="function")
+def provider_with_special_characters(provider):
+    """
+    Adds a special character sequence to extra_vars on a AWX provider to test BZ 1819310.
+    Note it seems the the AWX version 3.4 cannot be patched. It is refusing the request.
+    Note this was making the CFME unable to load the provider, thus this may interfere with other
+    tests when CFME is broken (due to regression).
+    """
+    template_name = provider.data['special_chars_template']
+    session = requests.session()
+    session.verify = False
+
+    extra_vars_with_special_chars = {'extra_vars': dedent(
+        r"""
+        ---
+        entry_path: "test\/21\\test"
+        """)}
+    empty_extra_vars = {'extra_vars': ""}
+
+    with closing(session):
+        url = provider.url
+        creds = credentials[provider.data.credentials]
+        session.auth = (creds['username'], creds['password'])
+        response = session.get(f'{url}/job_templates/', params=dict(name=template_name))
+        assert response.ok
+        template, = response.json()['results']
+
+        response = session.patch(f'{url}/job_templates/{template["id"]}/',
+                                 json=extra_vars_with_special_chars)
+        assert response.ok
+
+        yield provider
+
+        response = session.patch(f'{url}/job_templates/{template["id"]}/',
+                                 json=empty_extra_vars)
+        assert response.ok
+
+
+@pytest.mark.meta(automates=[1819310])
+@pytest.mark.provider(gen_func=providers,
```

Comment:
I think this mark, with the `required_fields` filter, should be decorating the fixture, not this test function.

The `provider_with_special_characters` fixture requires the `special_chars_template` field on the provider, and its current mark does not ensure that.

This probably functions fine - but any other test using `provider_with_special_characters` would have to include this provider mark as well.

I think that the mark can be simplified as well, since `providers_by_class` default gen func should work.

```
@pytest.mark.provider([AnsibleTowerProvider], selector=ONE_PER_VERSION, required_fields=['special_chars_template'])
```

Reply:
It seems it doesn't work when the marker is only on the fixture. It has to be on the test. You can take look on the tests history and commits I made.